x/nftmarket/keeper/nft_listed_class.go

```diff
@@ -4,18 +4,10 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/UnUniFi/chain/x/nftmarket/types"
+	"github.com/UnUniFi/chain/x/nftmint/keeper"
 )
 
-func (k Keeper) UpdateListedClass(ctx sdk.Context, listing types.NftListing) {
```

Comment:
do not remove this method.
Call DeleteListingFromListedClass in UpdateListedClass.
In the nft_listing file, I want to just call UpdateListedClass of nft_listed_class

Reply:
It's imossible to trigger the delete method based on the Listing.state. parm value because it actually does change that status only in the `DeleteNftListing` method in a nft_listing file.
That's why I summed up these `DeleteNftListing` and `DeleteListingFromListedClass`, instead of including deleting method in `UpdateListedClass` method.
If you want to do this, you have to change the code in nft_listing file.
What do you think about this?

Reply:
@mkXultra 
Fixed it.
Please review!

Reply:
@mkXultra 
Fixed it with `SetNftListings` and `DeleteNftListings`.
Please check it.